packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -227,7 +227,7 @@ function TabLine(props: TabLineProps) {
       setStyle(styleObj);
     }
 
-  }, [direction, setStyle, selectedTab, orientation, scale, selectedKey]);
+  }, [direction, setStyle, selectedTab, selectedTab?.offsetWidth, selectedTab?.offsetLeft, orientation, scale, selectedKey]);
```

Comment:
I think this may not update in all cases, only if we get lucky that the size update is tied to a state update. I think what we really want to do is attach a resizeObserver

An example of this changing outside of a state update is a font change

Reply:
Ahh, yep. I do remember seeing a comment on the issue about it not updating after a font change or FOUT type of thing, I just forgot to solve for it.

I'll take a look at the `useResizeObserver` util and se what I can do 👍

Reply:
@snowystinger Okay, this should be more resilient now. Moved the main functionality into a callback, and pass that to an observer as well as keeping a `useLayoutEffect` to catch the regular state changes.

Reply:
Thanks, will look again soon. We're in the midst of release planning and some other things at work, so apologies if it takes a few days